pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,278 @@
+// 1. get a list of jobs from big query that meet the passed criteria: platform, network etc.
+// 2. finds jobruns for matching jobs for the specified payload tag
+// 3. reads all junit for the each jobrun
+// 4. constructs a synthentic junit that check minimum required succuessful passes for configured tests
+type JobRunTestCaseAnalyzerOptions struct {
+	payloadTag          string
+	workingDir          string
+	jobRunStartEstimate time.Time
+	clock               clock.Clock
+	timeout             time.Duration
+	platform            string
+	infrastructure      string
+	network             string
+	ciDataClient        jobrunaggregatorlib.CIDataClient
+	ciGCSClient         jobrunaggregatorlib.CIGCSClient
+	gcsClient           *storage.Client
+	requiredPasses      []requiredPasses
+	payloadInvocationID string
+	jobGCSPrefixes      *[]jobGcsPrefix
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) getJobListFromNames(jobNames []string, allJobs []jobrunaggregatorapi.JobRow) ([]jobrunaggregatorapi.JobRow, error) {
+	jobMap := map[string]jobrunaggregatorapi.JobRow{}
+	for i := range allJobs {
+		job := allJobs[i]
+		jobMap[job.JobName] = job
+	}
+	jobs := []jobrunaggregatorapi.JobRow{}
+	for _, jobName := range jobNames {
+		if job, ok := jobMap[jobName]; !ok {
+			return jobs, fmt.Errorf("invalid job %s", jobName)
+		} else {
+			jobs = append(jobs, job)
+		}
+	}
+	return jobs, nil
+}
+
+func getJobInfrastructure(name string) string {
+	if strings.Contains(name, "upi") {
+		return "upi"
+	}
+	return "ipi"
+}
+
+// GetRelatedJobRuns gets all related job runs for analysis
+func (o *JobRunTestCaseAnalyzerOptions) GetRelatedJobRuns(ctx context.Context) ([]jobrunaggregatorapi.JobRunInfo, error) {
+	var jobRunsToReturn []jobrunaggregatorapi.JobRunInfo
+	jobs, err := o.ciDataClient.ListAllJobs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get jobs: %w", err)
+	}
+	// if PR payload, only find the exact jobs
+	if len(o.payloadInvocationID) > 0 {
+		jobNames := []string{}
+		for i := range *o.jobGCSPrefixes {
+			jobGcsPreix := (*o.jobGCSPrefixes)[i]
+			jobNames = append(jobNames, jobGcsPreix.jobName)
+		}
+		jobs, err = o.getJobListFromNames(jobNames, jobs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get jobs from names: %w", err)
+		}
+	}
+
+	// For release-controller generated payload, this contains all the jobs in DB
+	// For PR payload, this list of jobs correspond to the list of jobGcsPreix passed
+	// from command args.
+	for i := range jobs {
+		job := jobs[i]
+		if (len(o.platform) != 0 && job.Platform != o.platform) ||
+			(len(o.network) != 0 && job.Network != o.network) ||
+			(len(o.infrastructure) != 0 && o.infrastructure != getJobInfrastructure(job.JobName)) {
+			continue
+		}
+
+		var jobRunLocator jobrunaggregatorlib.JobRunLocator
+
+		if len(o.payloadTag) > 0 {
+			jobRunLocator = jobrunaggregatorlib.NewPayloadAnalysisJobLocatorForReleaseController(
+				job.JobName,
+				o.payloadTag,
+				false,
```

Comment:
this is unreadable.  either assign to variable or comment or use a different method name.